lib/shoryuken/polling.rb

```diff
@@ -102,5 +123,94 @@ def queue_weight(queues, queue)
         queues.count { |q| q == queue }
       end
     end
+
+    class StrictPriority < BaseStrategy
+
+      def initialize(queues)
+        # Mapping of queues to priority values
+        @queue_priorities = queues
+          .each_with_object(Hash.new(0)) { |queue, h| h[queue] += 1 }
+
+        # Priority ordering of the queues
+        @queue_order = @queue_priorities
+          .to_a
+          .sort_by { |queue, priority| -priority }
+          .map(&:first)
+
+        # Pause status of the queues
+        @queue_status = @queue_order
```

Comment:
I think we can compress queue_status in just one field, something like "activation time". When `Time.now < activation_time`, queue is not active yet. When `Time.now >= activation_time`, queue is active.